scrapy/spidermiddlewares/urllength.py

```diff
@@ -37,4 +38,7 @@ def _filter(request):
             else:
                 return True
 
-        return (r for r in result or () if _filter(r))
+        if self.maxlength == default_settings.URLLENGTH_LIMIT:
+            return result
+        else:
+            return (r for r in result or () if _filter(r))
```

Comment:
We do not need to change anything here. The whole middleware will be disabled if the value is 0, so this code will not even be reached in that case.

Reply:
A few questions here: 
- When you talk about disabling the middleware, you mean doing something like the code below right?
 ```python
settings.getdict('SPIDER_MIDDLEWARES')[spider_middleware_class] = None
```
Note: In this particular case spider_middleware_class == 'scrapy.spidermiddlewares.urllength.UrlLengthMiddleware'

- On the other hand, as a consequence of this change the expected behavior will be a `None` value when we use `from_settings(cls, settings)` method right? I am asking this for testing purposes.

Reply:
When you raise `NotConfigured` during the component initialization (i.e. `from_settings`, `from_crawler` or `__init__`), the component is disabled. The effect is the same as the code you wrote there, setting the class to `None` on the corresponding setting, but users do not need to do that. In other words, with your change settting `MAX_URLLENGTH` to `0` becomes an alternative way to achieve the same effect as setting that middleware class to `None` in the corresponding setting.